### PR TITLE
AMM-1072 In table search changes

### PR DESCRIPTION
**Pull request by devikasuresh20**
## 📋 Description

JIRA ID: AMM-1072

Resolved in table search issue by adding proper data source and updating paginator after each search filtration.

---

## ✅ Type of Change

- [x] 🐞 **Bug fix** (non-breaking change which resolves an issue)
- [ ] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🔥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 **Refactor** (change that is neither a fix nor a new feature)
- [ ] ⚙️ **Config change** (configuration file or build script updates)
- [ ] 📚 **Documentation** (updates to docs or readme)
- [ ] 🧪 **Tests** (adding new or updating existing tests)
- [ ] 🎨 **UI/UX** (changes that affect the user interface)
- [ ] 🚀 **Performance** (improves performance)
- [ ] 🧹 **Chore** (miscellaneous changes that don't modify src or test files)

---

## ℹ️ Additional Information

Please describe how the changes were tested, and include any relevant screenshots, logs, or other information that provides additional context.

## Release Notes

- **New Features**
	- Enhanced filtering functionality across various components, ensuring pagination is consistently applied regardless of search terms.
	- Added user feedback messages for "No Records Found" in multiple components to improve user experience.
	- Introduced new validation checks and improved error handling in several forms.

- **Bug Fixes**
	- Fixed paginator assignments in various components to ensure correct pagination behavior during filtering operations.

- **Documentation**
	- Updated comments and logging for better clarity and debugging.

- **Style**
	- Improved styling for user feedback messages and adjusted layout elements for better visibility and usability.

### CODE REVIEW

File: src/app/app-provider-admin/provider-admin/configurations/wrapup-time-configuration/wrapup-time-configuration.component.ts [hunk 1]
```diff
@@ -210,6 +218,7 @@ export class WrapupTimeConfigurationComponent implements OnInit {
   givePrivilegeForWrapupTime(event: any, role: any) {
     const formArray: any = this.wrapupTimeForm.controls['timings'] as FormArray;
     this.dataSource.data = formArray;
+    this.dataSource.paginator = this.paginator;
```
[the remark this answers is not in the document]
Comment (devikasuresh20) [reply src/app/app-provider-admin/provider-admin/configurations/wrapup-time-configuration/wrapup-time-configuration.component.ts:221]: In this case, the paginator reassignment is a simple operation and creating additional method adds unnecessary abstraction with minimal benefits of reusability.

File: src/app/app-provider-admin/super-admin/super-admin/Activities/service-provider-master/service-provider-master.component.html [hunk 2]
```diff
@@ -113,6 +113,13 @@ <h4>
           Create New Provider
         </button>
       </div>
+      <div
+        class="pull-center"
+        style="margin-top: 35px; text-align: center"
+        *ngIf="filteredsearchResult.data.length === 0"
+      >
+        <strong>No Records Found</strong>
+      </div>
```
[the remark this answers is not in the document]
Comment (devikasuresh20) [reply src/app/app-provider-admin/super-admin/super-admin/Activities/service-provider-master/service-provider-master.component.html:122]: Retaining the existing implementation for displaying message "No records found"

File: src/app/app-provider-admin/provider-admin/activities/parking-place-master/parking-place-master.component.ts [hunk 3]
```diff
@@ -396,7 +396,8 @@ export class ParkingPlaceComponent implements OnInit, AfterViewInit {
   }
   filterComponentList(searchTerm?: string) {
     if (!searchTerm) {
-      this.filteredavailableParkingPlaces = this.availableParkingPlaces;
+      this.filteredavailableParkingPlaces.data = this.availableParkingPlaces;
+      this.filteredavailableParkingPlaces.paginator = this.paginator;
```
[the remark this answers is not in the document]
Comment (devikasuresh20) [reply src/app/app-provider-admin/provider-admin/activities/parking-place-master/parking-place-master.component.ts:400]: Retaining manual filtering implementation for now as this approach provides better control over filtering logic.

File: src/app/app-provider-admin/provider-admin/activities/work-location-mapping/work-location-mapping.component.ts [hunk 4]
```diff
@@ -1979,6 +1978,7 @@ export class WorkLocationMappingComponent implements OnInit {
           }
         }
       });
+      this.filteredmappedWorkLocationsList.paginator = this.paginator;
```
[the remark this answers is not in the document]
Comment (devikasuresh20) [reply src/app/app-provider-admin/provider-admin/activities/work-location-mapping/work-location-mapping.component.ts:1981]: Retaining manual filtering implementation for now as this approach provides better control over filtering logic.

File: src/app/app-provider-admin/provider-admin/activities/parking-place-master/parking-place-master.component.ts [hunk 5]
```diff
@@ -229,7 +229,7 @@
     this.showTableFlag = true;
     this.editable = false;
     this.createButton = true;
-    this.availableParkingPlaces.data = response.data;
+    this.availableParkingPlaces = response.data;
```
[the remark this answers is not in the document]
Comment (devikasuresh20) [reply src/app/app-provider-admin/provider-admin/activities/parking-place-master/parking-place-master.component.ts:232]: The current implementation directly assigns the API response to availableParkingPlaces without strict type checking.

File: src/app/app-provider-admin/provider-admin/activities/specialist-mapping/specialist-mapping.component.ts [hunk 6]
```diff
@@ -269,6 +269,7 @@ export class SpecialistMappingComponent implements OnInit {
   filterSpecializationList(searchTerm?: string) {
     if (!searchTerm) {
       this.filteredspecializationList.data = this.specializationList;
+      this.filteredspecializationList.paginator = this.paginator;
```
[the remark this answers is not in the document]
Comment (devikasuresh20) [reply src/app/app-provider-admin/provider-admin/activities/specialist-mapping/specialist-mapping.component.ts:272]: The paginator assignment is intentionally retained in both the branches. This ensures paginator is always reset , regardless of the filtration logic.

**Review by helenKaryamsetty — APPROVED** [review]

**Pull request merged**